packages/server/lib/cloud/api/put_fetch.ts

```diff
@@ -0,0 +1,77 @@
+import crossFetch from 'cross-fetch'
+import { NetworkError } from './network_error'
+import { HttpError } from './http_error'
+import { ParseError } from './parse_error'
+import { agent } from '@packages/network'
+
+type PutInit = Omit<RequestInit, 'agent' | 'method'>
+
+export const ParseKinds = Object.freeze({
+  JSON: 'json',
+  TEXT: 'text',
+})
+
+type ParseKind = typeof ParseKinds[keyof typeof ParseKinds]
+
+type PutOptions = PutInit & {
+  parse?: ParseKind
+}
+
+export async function putFetch <
+  TReturn extends any
+> (input: RequestInfo | URL, options: PutOptions = { parse: 'json' }): Promise<TReturn> {
+  const {
+    parse,
+    ...init
+  } = options
+
+  try {
+    const response = await (crossFetch as typeof crossFetch)(input, {
+      ...(init || {}),
+      method: 'PUT',
+      // cross-fetch thinks this is in the browser, so declares
+      // types based on that rather than on node-fetch which it
+      // actually uses under the hood. node-fetch supports `agent`.
+      // @ts-expect-error
+      agent,
```

Comment:
using node-fetch directly would probably remove the need for this @ts-expect-error